mrmustard/lab_dev/samplers.py

```diff
@@ -217,24 +217,30 @@ def probabilities(self, state, atol=1e-4):
         )
         return self._validate_probs(probs, atol)
 
-    def sample(self, state: State, n_samples: int = 1000, seed: int | None = None) -> np.ndarray:
-        initial_mode = state.modes[0]
-        initial_samples, probs = self.sample_prob_dist(state[initial_mode], n_samples, seed)
+    def sample_prob_dist(
```

Comment:
Looking at the implementation why override this? It's essentially the same implementation as the base minus returning the probabilities which I don't imagine is adding too much overhead

Reply:
Huh, thats super weird considering its not using TF. Could it have been the bug you solved above where `.probabilities` was called twice?

Reply:
If it really is adding so much overhead then what we could do is just add a `ret_probs` parameter on the base `sample_prob_dist`. Avoids having to override + duplicate code

Reply:
okay 👍  (and I don't know why it was adding so much overhead in the TF backend)